odl/tomo/geometry/conebeam.py

```diff
@@ -0,0 +1,557 @@
+        return self._motion_grid
+
+    @property
+    def angle_offset(self):
+        """Offset to the rotation angle in the azimuthal plane. Does not
+        imply an offset in z-direction. The actual angles then reside
+        within `angle_offset` + `angle_intvl`."""
+        return self._motion_params_offset
+
+    @property
+    def src_radius(self):
+        """Source circle radius of this geometry."""
+        return self._src_radius
+
+    @property
+    def det_radius(self):
+        """Detector circle radius of this geometry."""
+        return self._det_radius
+
+    @property
+    def ndim(self):
+        """Number of dimensions of this geometry."""
+        return 3
+
+    @property
+    def axis(self):
+        """The axis of rotation.
+
+        Returns
+        -------
+        axis : `numpy.ndarray`, shape ``(3,)``
+            The rotation axis
+        """
+        axis = self._axis
+        if axis is None:
+            axis = np.array([0, 0, 1])
+        elif isinstance(axis, (int, float)):
+            ind = axis
+            axis = np.zeros(3)
+            axis[ind] = 1
+        elif len(axis) == 3:
+            axis = np.array(axis)
+        else:
+            raise ValueError('`axis` intializer {} has wrong format'.format(
+                axis))
+
+        return axis
+
+    def det_rotation(self, angle):
+        """The detector rotation function.
+
+        Returns the matrix for rotating a vector in 3d by an angle ``angle``
+        about the rotation axis given by the property `axis` according to
+        the right hand rule.
+
+        The matrix is computed according to
+        `Rodrigues' rotation formula
+        <https://en.wikipedia.org/wiki/Rodrigues'_rotation_formula>`_.
+
+        Parameters
+        ----------
+        angle : `float`
+            The motion parameter. It must be contained in this
+            geometry's `motion_params`.
+
+        Returns
+        -------
+        rot_mat : `numpy.matrix`, shape ``(3, 3)``
+            The rotation matrix mapping the standard basis vectors in
+            the fixed ("lab") coordinate system to the basis vectors of
+            the local coordinate system of the detector reference point,
+            expressed in the fixed system.
+        """
+
+        angle = float(angle)
+        if angle not in self.motion_params:
+            raise ValueError('angle {} is not in the valid range {}.'
+                             ''.format(angle, self.motion_params))
+
+        axis = self.axis
+
+        cross_mat = np.matrix([[0, -axis[2], axis[1]],
+                               [axis[2], 0, -axis[0]],
+                               [-axis[1], axis[0], 0]])
+        dy_mat = np.asmatrix(np.outer(axis, axis))
+        id_mat = np.asmatrix(np.eye(3))
+        cos_ang = cos(angle)
+        sin_ang = sin(angle)
+
+        return cos_ang * id_mat + (1. - cos_ang) * dy_mat + sin_ang * cross_mat
```

Comment:
In some way yes. If it's not normalized, you get a scaling and a rotation, which is not what the function claims it does.